foreman_remote_execution 0.1 manual (Iteration 3)

stbenjam opened the pull request:
- [x] DEB instructions
- [x] Document change in default strategy https://github.com/theforeman/foreman_remote_execution/pull/65
- [x] SSH keys in parameter, provided snippet https://github.com/theforeman/foreman_remote_execution/pull/55
- [x] ~~Reccuring jobs? if it makes it in https://github.com/theforeman/foreman_remote_execution/pull/44~~
- [x] Customized SSH users https://github.com/theforeman/foreman_remote_execution/pull/69

stbenjam committed:
0.1.0 Manual

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+
+The whole remote execution system is pluggable, making it easy to add
+more providers of communication later. Therefore it's split into
+several components that must be installed on Foreman  and Foreman proxy.
```
double space _Foreman and_

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+packaged for. Since both Foreman and Smart proxy plugins requires
+Dynflow, Ruby 1.9+ is required.
+
+Foreman 1.10+ is compatible with 0.1.* versions of the plugin.
```
For discovery we have a matrix of versions, maybe to follow the same pattern?

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+## 2.1 Plugin Installation
+
+<div class="alert alert-info">
+The Smart proxy component requires a system ruby of 1.9 or higher.  Currently, this means the proxy can't be installed on CentOS 6/RHEL 6 or Ubuntu 12.04.
```
Double space

stbenjam replied:
Occasionally people notice I do this, double spacing after a  period is how I was taught and not a habit I'm going to easily break.  HTML condenses it into 1 anyway.

stbenjam replied:
(I'll sed replace them all if you really care about it)

domcleal replied:
You'll probably find this in all of my theforeman.org commits too... an unbreakable habit.

lzap replied:
> Occasionally people notice I do this, double spacing after a  period is how I was taught [1] and not a habit I'm going to break.  HTML condenses it into 1 anyway.
> 
> [1] Many people were, http://www.quickanddirtytips.com/education/grammar/two-spaces-after-a-period

Interesting, we haven't been ever told during our lessons here. I was
taking mine around 1996, so it's surprise to me. Our teacher was too
modern I guess ;-)

## 

Later,
 Lukas #lzap Zapletal

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+    foreman-installer --enable-foreman-plugin-remote-execution\
+                      --enable-foreman-proxy-plugin-remote-execution-ssh
+
+The installer will automatically configure an SSH key for you, or you may configure an existing see.  See the foreman-installer --help for more information.
```
Ditto

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+</div>
+
+This chapter covers clean Foreman and plugin installation. We assume smart
+proxy being installed on the same host, however it's not a requirement.  Please
```
Ditto

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+Alternatively, the Foreman proxy publishes the public key it over the
+API, so that one can just download it into authorized keys:
+
+    curl https://myproxy.example.com:8443/ssh/pubkey >> ~/.ssh/authorized_keys
```
Is this really Foreman or Smart-proxy API? Looks like Smart-proxy. I'd rather use "Foreman Proxy" in the text above, each of these can have a different key.

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+
+1. find and input with name motd
+2. determine its type
+3. based on type, fetch the value
```
Having my first read, I don't understand what the "type" is. Explain above please, I think "We support four types of inputs" is not enough.

stbenjam replied:
This isn't really mine, the reason there's two commits is so you view the diff of what this PR changes from 0.0 manual.  I'll fix it I guess...

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+interaction. Since the template is always rendered for specific host
+on which the job is executed we load the value this host data. These
+data can be either Fact (coming from Puppet/Salt/Chef), smart class
+parameter (when you use Foreman as Puppet/Salt ENC), smart variable or
```
Maybe to be more generic instead naming individual configuration management tools? Ansible is coming up and this can get out of date :-)

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+
+You can find all Jobs when you navigate to Monitor > Jobs. A table
+lists all jobs from history to future. You can search jobs by **Job
+name** with a queries like
```
Ok, the second person pronouns are a bit ugly in this section and above :-)

lzap commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,394 @@
+In order to perform the job on a host, there needs to be a Smart proxy
+available to do the job. In the default install, it was use all Smart proxies
+with a remote execution provider.  This can be turned off by setting
+**remote_execution_global_proxy** to false.
```
Global setting?

stbenjam commented:
Thanks @lzap, updated with your feedback.  I've also squashed the commits since so much changed, the diff was unreadable anyway.

Still more to add before we release iteration 3.

lzap commented:
> Still more to add before we release iteration 3.

Let me know once this is ready for another read.

## 

Later,
 Lukas #lzap Zapletal

iNecas commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,407 @@
+## 4.1 Determining the smart proxy for host
+
+In order to perform the job on a host, there needs to be a Smart Proxy
+available to do the job. In the default install, it was use all Smart proxies
```
`s/it was/it would/`. Also noting, that this happens unless some proxy is attached through a network interface would be good

iNecas commented:
Minor comments, otherwise good to go for me (but let's wait if we get the recurring PR in)

stbenjam commented:
Updated

iNecas commented:
ACK - pending one comment

iNecas commented on plugins/foreman_remote_execution/0.1/index.md:
```diff
@@ -0,0 +1,407 @@
+
+You can install both the Foreman and Smart Proxy plugin with the installer:
+
+    foreman-installer --enable-foreman-plugin-remote-execution\
```
When testing this, I've noticed `foreman-tasks` service was not running, it seems we need `--enable-foreman-plugin-tasks` here as well

stbenjam replied:
Hmm, good catch, updated, and fixed over here for next release https://github.com/theforeman/puppet-foreman/pull/375

stbenjam commented:
Updated, I think this is ready to go

mmoll commented:
I changed compatability -> compatibility in two lines, otherwise it was fine. Thanks @stbenjam!

stbenjam commented:
Thanks!
